modules/random.js

```diff
@@ -1,8 +1,35 @@
 module.exports.dice = function(message){
   var args = message.content.slice(5);
-  var diceValue = parseInt(args) || 20;
-  var roll = Math.floor(Math.random() * diceValue) + 1;
-  message.reply(`you rolled a ${roll}`);
+  var d = args.indexOf('d');
+
+  if(d >= 0)
+  {
+    var dice = parseInt(args.slice(0, d)) || 1;
+    var eyes = parseInt(args.slice(d + 1)) || 6;
+
+    // If we use a regular array here the values will be interpreted as strings for some reason, which causes problems when sorting.
+    var values = new Uint32Array(dice);
+    var sum = 0;
+
+    for(var i = 0; i < dice; ++i)
+    {
+      var e = Math.floor(Math.random() * eyes) + 1;
+      sum += e;
+      values[i] = e;
+    }
+
+    values.sort();
+    values.reverse();
+
+    message.channel.send(`${message.author} rolled ${sum} (${values.join(', ')})`);
```

Comment:
You don't need to change this, but check out Message.reply in the discord.js library.